### PR TITLE
Feat/implement language switch

### PR DESCRIPTION
**Pull request by transcendence12**
create i18n.config file for define locales,
add enum for LocaleSwitcher in I18nNamespaces,
add key-value in messages in en and pl json files to handle locale switcher translations,
create Locale Switcher folder for components and styles,
create LocaleSwitcher and LocaleSwitcherSelect for language selection on desktop and mobile devices,
reorganize code in Header.tsx and in HamburgerMenuMobileBtn.tsx in which is placed Locale Switcher

### CODE REVIEW

File: apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { I18nNamespaces } from '@/i18n'
+import { useLocale, useTranslations } from 'next-intl'
+import { LocaleSwitcherSelect } from './LocaleSwitcherSelect'
+
+// export const LocaleSwitcher = () => {
```
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcher.tsx:5]: **nitpick**: Consider removing commented-out code if it's no longer needed, or add a comment explaining why it's kept.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcherSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
+'use client'
+
+import { useMediaQuery } from '@/hooks/useMediaQuery'
+import useOutsideClick from '@/hooks/useOutsideClick'
```
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcherSelect.tsx:4]: **suggestion**: Ensure that the useOutsideClick hook is necessary and functioning as expected. If not, consider removing it.

**Review by NerdbordMentor — APPROVED** [review]
## Overall Feedback

The implementation of the language switcher feature is well-executed, providing a seamless way to toggle between English and Polish on both mobile and desktop platforms. The code is organized and follows a logical structure, with the necessary components and styles created for the Locale Switcher. The use of i18n for handling translations is appropriate and aligns with best practices for internationalization.

### Things Missing:

- [] Ensure that all commented-out code is either removed or justified with a clear reason for its presence.

### Things That Went Well:

- The integration of the Locale Switcher into existing components like Header and HamburgerMenuMobileBtn is done cleanly and efficiently.
- The use of SCSS for styling the Locale Switcher components is well-structured and maintains consistency with the rest of the project.
- The addition of new icons (ChevronIcon and GlobeIcon) enhances the user interface and provides a clear visual indication of the language selection feature.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcherSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  return (
+    <div
+      className={`${styles.container} ${isOpen ? styles.open : ''}`}
+      ref={dropdownRef}
+    >
+      <p className={styles.srOnly}>{t('label')}{label}</p>
+      <button
+        className={styles.button}
+        onClick={handleDropdown}
+        style={{
```
Comment (Sumick) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcherSelect.tsx:77]: **nit:** We should not use inline styles this way - if you need to make dynamic classes, we use `cx` method for that. Check out our other components how it's already implemented

File: apps/platform-app/src/components/LoginBtn/LoginBtnsWrapper.tsx [hunk 4]
```diff
@@ -5,8 +5,8 @@ const LoginBtnsWrapper = () => {
   return (
     <div className={styles.wrapper}>
       <LoginBtn variant="secondary">Login</LoginBtn>
-      <div className={styles.btnSeparator}></div>
-      <LoginBtn variant="secondary">Join as Hunter</LoginBtn>
+      {/* <div className={styles.btnSeparator}></div> */}
```
Comment (Sumick) [inline apps/platform-app/src/components/LoginBtn/LoginBtnsWrapper.tsx:8]: **nit:** Why removed / commented?

File: apps/platform-app/src/i18n.ts [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
 
-const locales = ['en', 'pl']
 
 export default getRequestConfig(async ({ locale }) => {
-  if (!locales.includes(locale)) {
-    // Tu można zwrócić domyślne tłumaczenia lub pusty obiekt
-    return { messages: {} }
-  }
+  // if (!locales.includes(locale)) {
+  //   // Tu można zwrócić domyślne tłumaczenia lub pusty obiekt
```
Comment (Sumick) [inline apps/platform-app/src/i18n.ts:8]: **nit:** To remove if we dont use it

File: apps/platform-app/src/components/LoginBtn/LoginBtnsWrapper.tsx (hunk 4 above)
Comment (transcendence12) [reply apps/platform-app/src/components/LoginBtn/LoginBtnsWrapper.tsx]: Beaucouse now we have new design in Figma.

**Review by Sumick — APPROVED** [review]
Looks and works very nice, good job Gosia!

File: apps/platform-app/messages/en.json [hunk 6]
```diff
@@ -1,4 +1,11 @@
-{
+{"LocaleSwitcher": {
```
Comment (Wiecek-K) [inline apps/platform-app/messages/en.json:1]: Format this code using Prettier

File: apps/platform-app/messages/pl.json [hunk 7]
```diff
@@ -1,4 +1,11 @@
-{
+{"LocaleSwitcher": {
```
Comment (Wiecek-K) [inline apps/platform-app/messages/pl.json:1]: Format this code using Prettier

File: apps/platform-app/src/app/[locale]/(profile)/(components)/Header/Header.tsx [hunk 8]
```diff
@@ -50,7 +51,11 @@ async function Header({ buttonsVariant = 'main' }: HeaderProps) {
     </li>
   ) : (
     <li>
-      <LoginBtn variant="tertiary">Login</LoginBtn>
+      <HamburgerMenuMobileBtn
+        userHasProfile={false} // tutaj ustawiam na false, żeby spełniało warunek dla niezalogowanych użytkowników
```
Comment (Wiecek-K) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/Header/Header.tsx:55]: comment in Polish

Comment (Wiecek-K) [reply apps/platform-app/src/app/[locale]/(profile)/(components)/Header/Header.tsx:55]: By the way, are hardcoded values safe here?

File: apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcher.tsx [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { I18nNamespaces } from '@/i18n'
+import { useLocale, useTranslations } from 'next-intl'
+import { LocaleSwitcherSelect } from './LocaleSwitcherSelect'
+
+export const LocaleSwitcher = () => {
+  const t = useTranslations(I18nNamespaces.LocaleSwitcher)
+  const locale = useLocale()
```
Comment (Wiecek-K) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/LocaleSwitcher/LocaleSwitcher.tsx:7]: If you use hooks inside a component, please make it a client component. "use client"

**Review by Wiecek-K — APPROVED** [review]
Good job

**Pull request merged**